pkg/common-controller/snapshot_controller.go

```diff
@@ -1186,6 +1237,49 @@ func (ctrl *csiSnapshotCommonController) getSnapshotClass(className string) (*cr
 	return class, nil
 }
 
+// getSnapshotDriverName is a helper function to get snapshot driver from the VolumeSnapshot.
+// We try to get the driverName in multiple ways, as snapshot controller metrics depend on the correct driverName.
+func (ctrl *csiSnapshotCommonController) getSnapshotDriverName(vs *crdv1.VolumeSnapshot) (string, error) {
+	klog.V(5).Infof("getSnapshotDriverName: VolumeSnapshot[%s]", vs.Name)
+
+	var driverName string
+	// Try to get from snapshot class first
+	if vs.Spec.VolumeSnapshotClassName != nil {
```

Comment:
I think we might need to swap the logic like following:

get driver from content.Spec.Driver
if content not found
get driver from snapshot class

There are two major reasons:
1. content.Spec.Driver is populated by controller in dynamic provisioning case. while snapshot class can be deleted and recreated, the new driver will not be propagated to the content, however in VolumeSnapshot it holds only a reference
2. in preprovisioned case, driver is a required field for content while there could be NO snapshot class.
WDYT?

Reply:
Agreed, moved this around to align with the above suggestion.